src/D2L.Bmx/Authenticate.cs

```diff
@@ -124,19 +126,39 @@ IOktaApi oktaApi
 		var currTime = DateTimeOffset.Now;
 		return sourceCache.Where( session => session.ExpiresAt > currTime ).ToList();
 	}
-
-	private static int PromptMfa( MfaOption[] mfaOptions ) {
-
-		Console.WriteLine( "MFA Required" );
+	private static int MatchDefaultMFAoption( MfaOption[] mfaOptions, string DefautMFAProvider, string DefaultMFAmethod ) {
 		for( int i = 0; i < mfaOptions.Length; i++ ) {
-			Console.WriteLine( $"[{i + 1}] {mfaOptions[i].Provider}: {mfaOptions[i].Name}" );
+			if( ( mfaOptions[i].Provider == DefautMFAProvider ) && ( mfaOptions[i].Name == DefaultMFAmethod ) ) {
+				return i;
+			}
```

Comment:
```suggestion
			return Array.IndexOf(mfaOptions, mfaOptions.FirstOrDefault(option => option.Provider == DefautMFAProvider && option.Name == DefaultMFAmethod))
```

[LINQ ](https://learn.microsoft.com/en-us/dotnet/standard/linq/)is pretty op once you start using it a bit. Not saying you should change what you have though

Reply:
sure, will use that